addons/test_mail_full/tests/test_web_push.py

```diff
@@ -373,3 +373,37 @@ def test_push_notification_regenerate_vpaid_keys(self):
                 partner_id=self.user_email.partner_id.id,
                 vapid_public_key=self.vapid_public_key,
             )
+
+    @patch.object(odoo.addons.mail.models.mail_thread.Session, 'post', return_value=SimpleNamespace(**{'status_code': 201, 'text': 'Ok'}))
+    def test_push_notifications_truncate_payload(self, post):
+        """
+        Ensure that when we send large bodies, the final encrypted data (post-encryption)
+        never exceeds 4096 bytes. We verify this via the 'data' argument in the patched
+        Session.post call.
+
+        - First send a 4096-character body, which should definitely cause truncation.
+        - Then send a 3900-character body, which may or may not need heavy truncation
+        but should remain safely under 4096 bytes once encrypted.
+        """
+
+        too_big_body = 'X' * 4096
```

Comment:
Would maybe test using some longer UTF-8 characters as well, to be sure.

Also tests could probably be implemented as a loop with subtests then

Reply:
@Ephoniz :wave: :up:  In case you missed the notif